pkg/deploy/util/util.go

```diff
@@ -416,3 +444,16 @@ func (s ByMostRecent) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s ByMostRecent) Less(i, j int) bool {
 	return !s[i].CreationTimestamp.Before(s[j].CreationTimestamp)
 }
+
+// ControllersByCreationTimestampAsc sorts a list of ReplicationControllers by creation timestamp oldest first, using their names as a tie breaker.
+type ControllersByCreationTimestampAsc []api.ReplicationController
```

Comment:
Looking at the logic for `DeploymentVersionFor`, what guarantees can we make about the versions returned? When would an RC have a malformed version annotation? Can we take those risks when cleaning up? Are we at risk of "cleanung up" the current deployment if someone's broken that annotation on it?